Update docs with adding a new halo2 circuit in 'mopro-wam'

sifnoc opened the pull request:
- Added more item in prerequisite page
- Describe how to add a new halo2 circuit implementation on `mopro-wasm`

vivianjeng commented on docs/docs/setup/web-wasm-setup.md:
````diff
@@ -28,3 +28,73 @@ For more details on building and integrating these circuits, refer to [`mopro-wa
+    The following command builds the wasm package in "mopro-wasm":
+
+   ```shell
+   mopro-wasm $ wasm-pack build --target web --out-name my-halo2-circuit
````
we should also let users use `mopro build` 🙏🏻

sifnoc replied:
I believe the contents of `rust-setup.md` and `web-wasm-setup.md` are intended for experienced users.
Therefore, I’ve left this section without incorporating the mopro cli. This is a mixed situation, and it might be better to cover how to update a customized circuit using the cli on a separate page.

It could be a good idea to introduce this while working on: https://github.com/zkmopro/mopro/issues/269

sifnoc commented on docs/docs/setup/web-wasm-setup.md:
```diff
@@ -28,3 +28,73 @@ For more details on building and integrating these circuits, refer to [`mopro-wa
+
+Follow these steps to update and build the user-defined circuit implementation for Wasm:
+
+1. **Update `Cargo.toml` in 'mopro-wasm'**
```
> I think the users should use `mopro-wasm-lib` (in `mopro-example-app`) instead of `mopro-wasm` in `zkmopro`.
 
For align consistancy with other setup doc, we should add create `mopro-wasm-lib` like "setup the rust project" in Rust setup and remove the "Proving from web browser" section :https://zkmopro.org/docs/setup/web-wasm-setup#proving-from-web-browser.
What do you think?

> Because in the future, the users don't need to clone the repo `zkmopro/mopro` and they can use CLI (I think we should support `mopro` binary for users to use mopro CLI like circom, they support [binaries](https://github.com/iden3/circom/releases/tag/v2.2.1) so they don't need to clone the repo and use the binary)

I totally agree. but currently only simple doc "Getting Started" for about how to setup with the mopro cli.

vivianjeng replied:
> For align consistancy with other setup doc, we should add create mopro-wasm-lib like "setup the rust project" in Rust setup and remove the "Proving from web browser" section :https://zkmopro.org/docs/setup/web-wasm-setup#proving-from-web-browser.
What do you think?

We should let user simply uses `mopro init`, `mopro build` for web
and he will find `mopro-wasm-lib` within the folder
(I realized that we should update react native setup and flutter setup, and guide them to look **Getting Started** section instead of **Rust Setup** section)
and yes we can remove the "Proving from web browser" section

sifnoc replied:
Looks good! I will update other setup docs that to rely on using `mopro cli` instead of starting with `Rust Setup` page.

sifnoc commented on docs/docs/setup/web-wasm-setup.md:
```diff
@@ -28,3 +28,73 @@ For more details on building and integrating these circuits, refer to [`mopro-wa
+
+2. **Create Wrapper Functions for Generate/Verify proof method**:
+
+   To compile Wasm code with the circuit, wrapper functions for generating and verifying proof methods in the user-defined circuit implementation must be created in `mopro-wasm/src/lib.rs`, using the example structure provided below:
```
Updated at [37f5b74](https://github.com/zkmopro/mopro/pull/292/commits/37f5b74c8fc0197e9fd53c68c9a7d7723ee44ad3)

sifnoc commented:
- Now the `Rust Setup` page renamed as `Manual Setup for Android/iOS Bindings` for accuracy.
- Moved the page item at the last in the side-bar. it looks more last option than using mopro cli
- Added setup Halo2-based rust project alongside with Circom-based rust project: which was only the content in the `Rust Setup`

vivianjeng commented on docs/docs/setup/flutter-setup.md:
```diff
@@ -1,6 +1,6 @@
 # Flutter Setup
 
-After completing the [Rust setup](rust-setup.md) and setting up either [iOS setup](ios-setup.md) or [Android setup](android-setup.md), you're ready to create a cross-platform project using [Flutter](https://flutter.dev/). <br/>
+Using ["Getting Started - 3. Mopro build"](/docs/getting-started.md#3-build-bindings) guide, you can generate the "MoproAndroidBindings" and "MoproIOSBindings" for the ios and android platforms in your project folder, you're ready to create a cross-platform project using [Flutter](https://flutter.dev/). <br/>
```
The official usage is `iOS`. We should keep this aligned in the paragraph.
```suggestion
Using ["Getting Started - 3. Mopro build"](/docs/getting-started.md#3-build-bindings) guide, you can generate the "MoproAndroidBindings" and "MoproiOSBindings" for the iOS and android platforms in your project folder, you're ready to create a cross-platform project using [Flutter](https://flutter.dev/). <br/>
```

sifnoc replied:
Thanks for picking up this.

vivianjeng commented on docs/docs/setup/react-native-setup.md:
```diff
@@ -1,6 +1,7 @@
 # React Native Setup
 
-After completing the [Rust setup](rust-setup.md) and setting up either [iOS setup](ios-setup.md) or [Android setup](android-setup.md), you're ready to create a cross-platform project using [React Native](https://reactnative.dev/). <br/>
+Using ["Getting Started - 3. Mopro build"](/docs/getting-started.md#3-build-bindings) guide, you can generate the "MoproAndroidBindings" and "MoproIOSBindings" for the ios and android platforms in your project folder. These bindings allow you to create a cross-platform project using [React Native](https://reactnative.dev/).<br/>
```
```suggestion
Using ["Getting Started - 3. Mopro build"](/docs/getting-started.md#3-build-bindings) guide, you can generate the "MoproAndroidBindings" and "MoproiOSBindings" for the iOS and android platforms in your project folder. These bindings allow you to create a cross-platform project using [React Native](https://reactnative.dev/).<br/>
```

vivianjeng commented on docs/docs/setup/react-native-setup.md:
````diff
@@ -10,9 +11,30 @@ In this tutorial, you will learn how to create a native Mopro module on both And
+mopro-example-app $ mopro create
+```
+
+Assuming you’ve successfully built the **ios** and **android** bindings: `MoproAndroidBindings` and `MoproIOSBindings` in your project folder, the mopro cli stored some parameters into the `Config.toml` file and reads them during the create command. It will also allow you to select the **react-native** template, as shown below:
````
```suggestion
Assuming you’ve successfully built the **iOS** and **android** bindings: `MoproAndroidBindings` and `MoproiOSBindings` in your project folder, the mopro cli stored some parameters into the `Config.toml` file and reads them during the create command. It will also allow you to select the **react-native** template, as shown below:
```

vivianjeng commented on docs/docs/setup/rust-setup.md:
```diff
@@ -168,3 +168,122 @@ CONFIGURATION=release cargo run --bin android # Release mode
+
+[dependencies]
+mopro-ffi = { git = "https://github.com/zkmopro/mopro.git", branch = "main" }
+rust-witness = "0.1.1"
```
actually we don't need `rust-witness` for halo2 circuits.
(We should update cli template as well)

vivianjeng commented on docs/docs/setup/rust-setup.md:
```diff
@@ -168,3 +168,122 @@ CONFIGURATION=release cargo run --bin android # Release mode
+
+[build-dependencies]
+mopro-ffi = { git = "https://github.com/zkmopro/mopro.git", branch = "main" }
+rust-witness = "0.1.1"
```
same here

vivianjeng commented on docs/docs/setup/web-wasm-setup.md:
````diff
@@ -1,30 +1,103 @@
+The wasm code can be imported and used in a web application, as illustrated below:
+
+```javascript
+const mopro_wasm = await import('./pkg/my-halo2-circuit.js');
````
does it change the name of the package?
I suppose the library will be called `mopro_wasm_lib.js` with `mopro build`?

sifnoc replied:
Ah, its left over. I fixed at [8693285](https://github.com/zkmopro/mopro/pull/292/commits/8693285140d906974070180700213377fc7d12d1)

vivianjeng reviewed (commented):
The current version looks good to me! Thank you!

can you also update the doc in `versioned_doc/version-0.1.0`?
It is confusing to have 2 docs 
but one is for current version (`0.1.0`) and the other is for the next version (might be`0.2.0`)
so just copy the current files in `docs/` and paste to `version-0.1.0/`

also the `mopro-wasm` should be copied to `docs/` too
Thank you

vivianjeng commented:
and there is an error in cloudflare
```sh

00:38:30.251 | [cause]: Error: Docusaurus found broken links!
-- | --
00:38:30.251 |  
00:38:30.251 | Please check the pages of your site in the list below, and make sure you don't reference any path that does not exist.
00:38:30.251 | Note: it's possible to ignore broken links with the 'onBrokenLinks' Docusaurus configuration, and let the build pass.
00:38:30.252 |  
00:38:30.252 | Exhaustive list of all broken links found:
00:38:30.252 | - Broken link on source page path = /docs/next/setup/react-native-setup:
00:38:30.252 | -> linking to docs/react-native-setup#option-1-use-mopro-cli (resolved as: /docs/next/setup/docs/react-native-setup#option-1-use-mopro-cli)
00:38:30.252 |  
00:38:30.252 | at throwError (/opt/buildhome/repo/docs/.yarn/cache/@docusaurus-logger-npm-3.2.1-11923e9c7b-9d5db5253e.zip/node_modules/@docusaurus/logger/lib/index.js:79:11)
00:38:30.252 | at reportBrokenLinks (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/server/brokenLinks.js:242:47)
00:38:30.252 | at handleBrokenLinks (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/server/brokenLinks.js:274:5)
00:38:30.253 | at executeBrokenLinksCheck (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/commands/build.js:182:47)
00:38:30.253 | at /opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/commands/build.js:136:66
00:38:30.253 | at Object.async (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/utils.js:36:47)
00:38:30.253 | at buildLocale (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/commands/build.js:136:30)
00:38:30.253 | at async tryToBuildLocale (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/commands/build.js:46:13)
00:38:30.253 | at async /opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/commands/build.js:64:9
00:38:30.253 | at async mapAsyncSequential (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-utils-virtual-daa5b7b8bd/0/cache/@docusaurus-utils-npm-3.2.1-cfade3be35-ea862b178e.zip/node_modules/@docusaurus/utils/lib/jsUtils.js:20:24)
00:38:30.253 | at async Command.build (/opt/buildhome/repo/docs/.yarn/__virtual__/@docusaurus-core-virtual-0694d71873/0/cache/@docusaurus-core-npm-3.2.1-027b2f9feb-9267f08b41.zip/node_modules/@docusaurus/core/lib/commands/build.js:62:5)
```

You can check by
```sh
cd docs && yarn build
```
to get the error

vivianjeng commented on docs/docs/setup/react-native-setup.md:
```diff
@@ -10,9 +11,30 @@ In this tutorial, you will learn how to create a native Mopro module on both And
+## Option 2: Clone the Repository and Import the Bindings
 
-### Option 1: Clone the Repository and Import the Bindings
+This option is a more manual compared to [Option 1](docs/react-native-setup#option-1-use-mopro-cli). You can clone a pre-configured repository and manually import the generated bindings into your React Native project.
```
You can just use hashtag to navigate within the page
```suggestion
This option is a more manual compared to [Option 1](#option-1-use-mopro-cli). You can clone a pre-configured repository and manually import the generated bindings into your React Native project.
```

sifnoc replied:
That’s why it keeps failing. I wasn’t aware that I was just running `yarn start instead` of `building` it. Thank you

sifnoc commented:
> The current version looks good to me! Thank you!
> 
> can you also update the doc in `versioned_doc/version-0.1.0`? It is confusing to have 2 docs but one is for current version (`0.1.0`) and the other is for the next version (might be`0.2.0`) so just copy the current files in `docs/` and paste to `version-0.1.0/`
> 
> also the `mopro-wasm` should be copied to `docs/` too Thank you

Synced at last two commits: 270f9df 93f0f4a

vivianjeng reviewed (approved):
LGTM!
Thank you so much!
